File: Best Time to Buy and Sell Stock - Leetcode 121/Best Time to Buy and Sell Stock - Leetcode 121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let minPrice = Infinity;
    let maxProfit = 0;

    for (const price of prices) {
        if (price < minPrice) {
            minPrice = price;
        }
        const profit = price - minPrice;
        if (profit > maxProfit) {
            maxProfit = profit;
        }
    }

    return maxProfit;
};

File: Best Time to Buy and Sell Stock - Leetcode 121/Best Time to Buy and Sell Stock - Leetcode 121.go
package solution

func maxProfit(prices []int) int {
    maxProf := 0
    min := prices[0]
    for i:=1;i<len(prices);i++{
        profit := prices[i] - min
        if maxProf < profit {
            maxProf = profit
        }
        if min > prices[i] {
            min = prices[i]
        }
    }
    return maxProf
}
